packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -639,9 +639,10 @@ function buildModuleSchema(
   const [moduleSpec] = moduleSpecs;
 
   if (moduleSpec.id.name !== 'Spec') {
-    throw new MisnamedModuleFlowInterfaceParserError(
+    throw new MisnamedModuleInterfaceParserError(
       hasteModuleName,
       moduleSpec.id,
+      'Flow',
     );
```

Comment:
The tasks was slightly different: this PR export just the type from the `index.js` to the `error-utils`.
However, this step was responsibility of #34896 (which this PR should depend on).

What I was expecting here, was a function like:
```
function throwIfModuleInterfaceIsMisnamed(nativeModuleName: string, module: ASTNode, parserType: ParserType) {
  if (moduleSpec.id.name !== 'Spec') {
    throw new error
  }
}
```

That will replace these 6 lines in the `flow/modules/index.js` and in the `typescript/modules/index.js`.

In this way:
1. if we need to update the logic that throws the error, we do it in a single place
2. the parser code is more readable and self documenting.

Could you update this with this suggestion?

Thank you so much! 🙏